### ISSUE TITLE
Screen Reader is not announcing about the hint (Press Ctrl K) to invoke the search button.

### ISSUE THREAD

**Issue by msftedad** [facebook/docusaurus#9167]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

**Test Environment: **    
OS: Windows 11, Version: 22H2, Build: 22621.1992   
Microsoft Edge Version: 114.0.1823.86 (Official build) (64-bit)
Screen Reader Used: Narrator

**Prerequisites: **  
Turn on Screen Reader   

**Steps to Reproduce: **    
1. Open https://www.fast.design/ URL.
2. Navigate to link 'Documentation' using tab key and invoke it using enter key.
3. Navigate to search button using tab key and invoke it using enter key.
4. Verify whether Screen Reader is announcing about the hint (Press Ctrl K) to invoke the search button or not.
 
**Actual result: **  
Screen Reader is not announcing about the hint (Press Ctrl K) to invoke the search button.

**Expected Results: **
Screen Reader should announce about the hint (Press Ctrl K) to invoke the search button when screen Reader focus moves on it.

Screen reader fails announcing about the hint (Press Ctrl K) to invoke the search button.
User is not able to know the provided hint.
Screen reader should announce hint (Press Ctrl K) ti invoke the search button.
User with visual impairment who rely on screen reader will not be known about the short key to invoke the search button.

### Steps to reproduce

1. Open https://www.fast.design/ URL.
2. Navigate to link 'Documentation' using tab key and invoke it using enter key.
3. Navigate to search button using tab key and invoke it using enter key.
4. Verify whether Screen Reader is announcing about the hint (Press Ctrl K) to invoke the search button or not. 
![Dev Tools](https://github.com/facebook/docusaurus/assets/93735775/4ad16bd1-ff47-49ef-84c0-63da5d7e8f80)

### Expected behavior

Screen Reader should announce about the hint (Press Ctrl K) to invoke the search button when screen Reader focus moves on it.

### Actual behavior

Screen Reader is not announcing about the hint (Press Ctrl + K) to invoke the search button.

### Your environment

- Public source code:
- Public site URL: https://www.fast.design/
- Docusaurus version used:
OS: Windows 11, Version: 22H2, Build: 22621.1992   
Microsoft Edge Version: 114.0.1823.86 (Official build) (64-bit)
Screen Reader Used: Narrator

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Hi, the search dialog is not maintained by us but by Algolia, so this issue should be reported to them.

cc @shortcuts

**Comment by msftedad** [comment]
Hi we have connected with po team and we got a below reply

 We have verified the issue in Algolia and observed we are not getting the similar kind of UI. For search Algolia no hint is defined. Please let us know correct product or UI to repro and file a new bug.  

Can you please check and let us know. Thanks.
